### DISCUSSION TITLE
Obfuscation performance

### DISCUSSION THREAD

**Discussion by doctor8296** [javascript-obfuscator/javascript-obfuscator#1206] [General]
We know the fact that most of the time the code itself start to works a bit slower after obfuscation. This is okay sometimes, when the obfuscated code is not very important part of the app that do some important calculations and actually can affect user experience.
But what about obfuscation itself? For very small piece of code which is like 7 sane JavaScript lines and with the following settings (very light imho): 
```js
{
        compact: true,
        simplify: true,
        identifierNamesGenerator: 'mangled-shuffled',
        renameGlobals: true,
        splitStrings: true,
        splitStringsChunkLength: 3,
        stringArray: true,
        stringArrayEncoding: ['rc4'],
        stringArrayShuffle: true,
        transformObjectKeys: true,
        unicodeEscapeSequence: true,
        seed: 0
}
```
it takes from 50 to 100 ms to obfuscate the source! 
And this is a lot if we talking about site loading performance, and it was just a little piece from the many others (that I have).

So my question is 
### can we improve the obfuscation speed?

**Comment by slig** [comment]
You should obfuscate in your build process, so it doesn't affect the loading times.

**Reply by doctor8296** [reply]
The thing is that I have very specific case where I need to reobfuscate code each request.

**Reply by slig** [reply]
Got it. Then I'm not sure how to deal with it, as the obfuscator is blazing fast for its purpose. Maybe try running it on Bun, compiling it to wasm?
